rayon-core/src/lib.rs

```diff
@@ -103,6 +103,20 @@ pub use self::thread_pool::current_thread_index;
 pub use self::thread_pool::ThreadPool;
 pub use self::thread_pool::{yield_local, yield_now, Yield};
 
+#[cfg(not(all(
+    target_arch = "wasm32",
+    target_os = "unknown",
+    target_feature = "atomics"
+)))]
+pub use std::sync;
+
+#[cfg(all(
+    target_arch = "wasm32",
+    target_os = "unknown",
+    target_feature = "atomics"
+))]
+pub use wasm_sync as sync;
```

Comment:
I definitely do not want to put these in the public API. If `rayon` needs to use it too, that should be a dependency on its own.

Reply:
Okay. I figured sharing might be easier, but I don't have strong opinion. I'll change when I'm back at computer.